### PR TITLE
Allow oscap-ssh to use custom SSH options

### PR DESCRIPTION
**Pull request by matejak**
This PR allows the caller to pass SSH connection options.
The use case for this is the SSG test suite - as containers or VMs have the same IP address but different SSH server certificates, one would be bothered by the `known_hosts` records that would cause false alarms.
This PR consists of some refactoring and backwards-compatible usage of the `SSH_ADDITIONAL_OPTIONS` environment variables for a list of SSH options.

### CODE REVIEW

File: utils/oscap-ssh [hunk 1]
```diff
@@ -80,61 +80,94 @@ function usage()
...
 shift 2
 
+check_oscap_arguments "$@"
+
 MASTER_SOCKET_DIR=$(mktemp -d)
 MASTER_SOCKET="$MASTER_SOCKET_DIR/ssh_socket"
 
 echo "Connecting to '$SSH_HOST' on port '$SSH_PORT'..."
-ssh -M -f -N -o ServerAliveInterval=60 -o ControlPath="$MASTER_SOCKET" -p "$SSH_PORT" "$SSH_HOST" || die "Failed to connect!"
+ssh -M -f -N -o ServerAliveInterval=60 -o ControlPath="$MASTER_SOCKET" -p "$SSH_PORT" $SSH_ADDITIONAL_OPTIONS "$SSH_HOST" || die "Failed to connect!"
```
Comment (cipherboy) [inline utils/oscap-ssh]: Could we perhaps add a wrapper for this command as well? I realize that it isn't reused and it adds more lines than necessary, but I think it'd be nice to have.

File: utils/oscap-ssh [hunk 2]
```diff
@@ -80,61 +80,94 @@ function usage()
...
+    elif [ "$1" == "info" ]; then
+        true
+    elif [ "$1 $2" == "xccdf eval" ]; then
+        true
+    elif [ "$1 $2" == "oval eval" ]; then
+        true
+    elif [ "$1 $2" == "oval collect" ]; then
+        true
+    else
+        die "This script only supports '-h', '--help', '--v', '--version', 'info', 'xccdf eval', 'oval eval' and 'oval collect'."
```
Comment (cipherboy) [inline utils/oscap-ssh]: I realize that the old version is wrong, but doesn't this technically support `--sudo` as well?

**Comment by matejak** [comment]
Thanks for review @cipherboy , I have updated the PR.

**Comment by matejak** [comment]
@matusmarhefka Good point, I have rebased and inserted the man page section in.

**Comment by matusmarhefka** [comment]
@matejak Great stuff, thank you, I am merging this.

**Pull request merged**